dali/operators/reader/loader/numpy_loader.cc

```diff
@@ -38,6 +37,75 @@ TypeInfo TypeFromNumpyStr(const std::string &format) {
   DALI_FAIL("Unknown Numpy type string");
 }
 
+void ParseHeaderMetadata(NumpyParseTarget& target, std::string header) {
+  header.erase(std::remove_if(header.begin(), header.end(), ::isspace), header.end());
+
+  const char t1[] = "{'descr':\'";
+  size_t l1 = strlen(t1);
```

Comment:
It's quite pointless to measure a constant string.
Alternatives:
1.
```
static const std::string t1 = "{'descr:'";
```
and use .length()
2.
```suggestion
size_t l1=sizeof(t1)-1;
```